packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -214,6 +217,28 @@ export class DiscoveryWatcher {
     }
   }
 
+  // 3rd party APIs are unstable, so we do a sanity check before sending
+  // notifications, which makes the same request again and compares the
+  // results.
+  async sanityCheck(
+    discovery: ProjectParameters,
+    projectConfig: DiscoveryConfig,
+    blockNumber: number,
+  ) {
+    const secondDiscovery = await this.discoveryEngine.run(
+      projectConfig,
+      blockNumber,
+    )
+
+    if (!isEqual(discovery, secondDiscovery)) {
```

Comment:
I think this makes sense, but I'd like to get a third opinion on whether this check doesn't run into some weird edge cases because for some reason our discovery code is non-deterministic @michalsidzej